### DISCUSSION TITLE
[NoDriver]How to locate elements inside nested iframes using nodriver

### DISCUSSION THREAD

**Discussion by MoonLite11** [ultrafunkamsterdam/undetected-chromedriver#2105] [Q&A]
Hi,

I’m using your nodriver library, and I’m trying to locate an element inside a structure like this:

iframe > iframe > element

I can access the first iframe and load its content_document, but when I try to navigate to the second iframe inside it, it doesn’t load its content_document, even though I can see it’s there. Additionally, the shadow-root is absent, and enabling the expert mode doesn’t help.

Could you provide some guidance on how to properly handle nested iframes with nodriver?

Thanks!

**Comment by thejoswamy** [comment]
Can you show how you are able to access first iframe and load its content_document

**Comment by MoonLite11** [comment]
`iframe = await page.select('iframe[id=token-credit-card-iframe]')` But when I try to find in this iframe, another one, it finds it with an empty content_document `iframe_cc = await iframe.query_selector_all('iframe[class=js-iframe]')`
